lib/Rex/Commands/File.pm

```diff
@@ -552,15 +558,6 @@ sub file {
     }
 
   }
-  elsif ( exists $option->{"source"} && !$is_directory ) {
-    $option->{source} =
-      Rex::Helper::Path::get_file_path( $option->{source}, caller() );
-
-    # HOOKS: for this case you have to use the upload hooks!
-    $__ret = upload $option->{"source"}, "$file", force => 1;
-
-    $need_md5 = 0 if $__ret->{changed} == 0;
```

Comment:
At first I wondered why `$need_md5` doesn't need to be set in the merged codepath. However, the merged code path turns out to have its own changed-ness detection (through md5s, but not through "need_md5". (thanks for answering my questions on IRC).